docs/rfcs/0561-list-metadata-reuse.md

```diff
@@ -0,0 +1,107 @@
+- Proposal Name: `list_metadata_reuse`
+- Start Date: 2022-08-23
+- RFC PR: [datafuselabs/opendal#561](https://github.com/datafuselabs/opendal/pull/561)
+- Tracking Issue: None
+
+# Summary
+
+Add a `metadata` field for `DirEntry`s.
+
+# Motivation
+
+Users may expect to browse metadata of some directories' child files and directories.
+`list()` seems to be an ideal way to complete this job. 
+
+Thus, they start iterating on it, but soon they realized the `DirEntry`, which could only offer the name (or path, 
+more precisely) and access mode of the object.
+
+So they have to call `meta()` for each name they extracted from the iterator.
+```
+// Pseudo code example of listing size of files
+for file in list().await {
+    let name = file.path
+    let size = object(name).meta().await.file_size()
+}
+```
+
+But...wait! many storage-services returns object metadata when listing, like HDFS, AWS and GCS. The rust standard library
+returns metadata when listing local file systems, too.
+
+In the previous versions of OpenDAL we just simply ignored them. This wastes users' time on requesting on metadata.
+
+# Guide-level explanation
+
+The pseudocode will be changed to the following code with this RFC:
```

Comment:
Please giving an example based on real code.

OpenDAL takes RFC seriously. We expect no ambiguity in the RFC.